packages/mobile/src/account/profileInfo.ts

```diff
@@ -0,0 +1,317 @@
+
+// export function* addMetadataClaim() {
+//   try {
+//     const contractKit = yield call(getContractKit)
+//     const account = yield select(currentAccountSelector)
+//     const metadata = IdentityMetadataWrapper.fromEmpty(account)
+//     yield call(
+//       [metadata, 'addClaim'],
+//       createStorageClaim(BUCKET_URL),
+//       NativeSigner(contractKit.web3.eth.sign, account)
+//     )
+//     Logger.info(TAG + '@addMetadataClaim' + 'created storage claim on chain')
+//     yield call(writeToGCPBucket, metadata.toString(), `${account}/metadata.json`)
+//     Logger.info(TAG + '@addMetadataClaim' + 'uploaded metadata.json')
+//     const accountsWrapper: AccountsWrapper = yield call([
+//       contractKit.contracts,
+//       contractKit.contracts.getAccounts,
+//     ])
+//     const setAccountTx = accountsWrapper.setMetadataURL(`${BUCKET_URL}${account}/metadata.json`)
+//     const context = newTransactionContext(TAG, 'Set metadata URL')
+//     yield call(sendTransaction, setAccountTx.txo, account, context)
+//     Logger.info(TAG + '@addMetadataClaim' + 'set metadata URL')
+//   } catch (error) {
+//     Logger.error(`${TAG}/addMetadataClaim`, 'Could not add metadata claim', error)
+//     throw error
+//   }
+// }
+
+export function* uploadNameAndPicture() {
+  const contractKit = yield call(getContractKit)
+  const account = yield select(currentAccountSelector)
+  const offchainWrapper = new UploadServiceDataWrapper(contractKit, account)
+
+  const name = yield select(nameSelector)
+  const nameAccessor = new PrivateNameAccessor(offchainWrapper)
+  let writeError = yield call([nameAccessor, 'write'], { name }, [])
+  Logger.info(TAG + 'uploadNameAndPicture', 'uploaded profile name')
+
+  const pictureUri = yield select(pictureSelector)
+  if (pictureUri) {
+    const data = yield call(RNFS.readFile, pictureUri, 'base64')
+    const mimeType = extensionToMimeType[pictureUri.split('.').slice(-1)] || 'image/jpeg'
+    const dataURL = getDataURL(mimeType, data)
+    const pictureAccessor = new PrivatePictureAccessor(offchainWrapper)
+    writeError = yield call([pictureAccessor, 'write'], dataURL, [])
+    Logger.info(TAG + 'uploadNameAndPicture', 'uploaded profile picture')
+  }
+
+  if (writeError) {
+    Logger.error(TAG + '@uploadNameAndPicture', writeError)
+    throw Error('Unable to write data')
+  }
+}
+
+// this function gives permission to the recipient to view the user's profile info
+export function* uploadSymmetricKeys(recipientAddresses: string[]) {
+  // TODO: check if key for user already exists, skip if yes
+  const account = yield select(currentAccountSelector)
+  const contractKit = yield call(getContractKit)
+  yield call(unlockDEK)
+  const offchainWrapper = new UploadServiceDataWrapper(contractKit, account)
+
+  const nameAccessor = new PrivateNameAccessor(offchainWrapper)
+  let writeError = yield call([nameAccessor, 'allowAccess'], recipientAddresses)
```

Comment:
Same issue here. Generally I prefer to return as early as possible from functions. With this the log line on 266 will still fire even though there could have been two errors.

Maybe something like this works a bit more explicitly

```
const nameAccessor = new PrivateNameAccessor(offchainWrapper)
const nameWriteError = yield call([nameAccessor, 'allowAccess'], recipientAddresses)
if (nameWriteError) {
  // log
  return // or throw
}

const pictureUri = yield select(pictureSelector)
if (pictureUri) {
  const pictureAccessor = new PrivatePictureAccessor(offchainWrapper)
  const pictureWriteError = yield call([pictureAccessor, 'allowAccess'], recipientAddresses)
  if (pictureWriteError) {
    // log
    return // or throw
  }
}

Logger.info(TAG + '@uploadSymmetricKeys', 'uploaded symmetric keys for ' + recipientAddresses)
```